bundles/ui/org.eclipse.smarthome.ui/src/main/java/org/eclipse/smarthome/ui/internal/items/ItemUIRegistryImpl.java

```diff
@@ -868,9 +868,9 @@ private boolean matchStateToValue(State state, String value, String matchConditi
                 logger.debug("matchStateToValue: Decimal format exception: ", e);
             }
         } else if (state instanceof DateTimeType) {
-            Calendar val = ((DateTimeType) state).getCalendar();
-            Calendar now = Calendar.getInstance();
-            long secsDif = (now.getTimeInMillis() - val.getTimeInMillis()) / 1000;
+            ZonedDateTime val = ((DateTimeType) state).getZonedDateTime();
+            ZonedDateTime now = ZonedDateTime.now();
+            long secsDif = (now.getNano() - val.getNano());
```

Comment:
seconds = nano - nano? I don't think that is correct?
Do you want to use e.g. `TimeUnit.NANOSECONDS.toSeconds(duration)`?

But why not using methods that are part of the framework?
`ChronoUnit.SECONDS.between(zonedDateTime1, zonedDateTime2);`